src/core/obj.js

```diff
@@ -254,6 +254,62 @@ class Catalog {
     return permissions;
   }
 
+  get optionalContentConfig() {
+    const properties = this.catDict.get("OCProperties");
+    if (!properties) {
+      return null;
+    }
+    const groupsData = properties.get("OCGs");
+    if (!Array.isArray(groupsData)) {
+      return null;
+    }
+    const groups = [];
+    const groupRefs = [];
+    // Ensure all the optional content groups are valid.
+    for (const groupRef of groupsData) {
+      if (!isRef(groupRef)) {
+        continue;
+      }
+      groupRefs.push(groupRef);
+      const group = properties.xref.fetchIfRef(groupRef);
+      groups.push({
+        id: groupRef.toString(),
+        name: group.get("Name"),
+        intent: group.get("Intent"), // TODO validate this.
+      });
+    }
+    const defaultConfig = properties.get("D");
+    if (!defaultConfig) {
+      return null;
+    }
+    const config = this._readOptionalContentConfig(defaultConfig, groupRefs);
+    config.groups = groups;
+    return config;
```

Comment:
Similar to all other methods here, see e.g. `get permissions()` above, please wrap this in `try...catch` to prevent breaking errors if data fetching fails (since Dict.get(...) calls *could* throw).

Reply:
Will `get optionalContentConfig()` always return the same data for a document with Optional Content groups?
If so, please `shadow` the return values to avoid having to repeatedly re-parsing this for *every* single page. (Or at least, please `shadow` the `return null;` cases.)